eng/scripts/Update-Packages.ps1

```diff
@@ -0,0 +1,160 @@
+
+###########
+# Imports #
+###########
+
+. $PSScriptRoot/Shared.ps1
+
+#############
+# Functions #
+#############
+
+Function Get-NewVersion
+{
+    Param(
+        [Parameter(Mandatory = $true)]
+        [ValidateNotNullOrWhiteSpace()]
+        [string]$ElementValue,
+        [Parameter(Mandatory = $true)]
+        [ValidateNotNullOrWhiteSpace()]
+        [int]$NegativePosition
+    )
+
+    If ($NegativePosition -ge 0)
+    {
+        Write-Error "The position number '$NegativePosition' should have been negative." -ErrorAction Stop
+    }
+
+    $components = $ElementValue -Split '\.'
+
+    If (($NegativePosition * -1) -gt $components.Length)
+    {
+        Write-Error "Version number '$ElementValue' is out of range." -ErrorAction Stop
+    }
+
+    $components[$NegativePosition] = [int]$components[$NegativePosition] + 1
+
+    Return [string]::Join('.', $components)
+}
+
+Function Get-AssemblyNameLowerCase
+{
+    Param(
+        [Parameter(Mandatory = $true)]
+        [ValidateNotNullOrWhiteSpace()]
+        [string]$MSBuildVersionFilePath
+    )
+
+    $assemblyName = [System.IO.Path]::GetFileNameWithoutExtension($MSBuildVersionFilePath)
+
+    If ($assemblyName -eq "Versioning")
+    {
+        $srcFolderPath = Join-Path -Path ([System.IO.Path]::GetDirectoryName($MSBuildVersionFilePath)) -ChildPath "src"
+        $actualCsproj = Get-ChildItem -Path $srcFolderPath -Filter "*.*proj" -Recurse | Select-Object -First 1
+        $assemblyName = [System.IO.Path]::GetFileNameWithoutExtension($actualCsproj.FullName)
+    }
+
+    Return $assemblyName.ToLowerInvariant()
+}
+
+Function Save-Project
+{
+    Param(
+        [Parameter(Mandatory = $true)]
+        [ValidateNotNullOrEmpty()]
+        [System.Xml.Linq.XDocument]
+        $XDoc,
+        [Parameter(Mandatory = $true)]
+        [ValidateNotNullOrWhiteSpace()]
+        [string]$File
+    )
+
+    # Ideally, we would've simply used:
+    #   $XDoc.Save($file, [System.Xml.Linq.SaveOptions]::DisableFormatting)
+    # Unforutnately, it does not work as expected in PowerShell. Unlike in C#, the xml declaration
+    # is still getting added at the top even though we specified to disable formatting.
+    # Workaround: Use an XmlWriter to serialize into a string the XML without the xml declaration.
+    # Also, we need to remove the trailing new line that is added by the XmlWriter.
+
+    $stringWriter = New-Object System.IO.StringWriter
+    $xmlWriterSettings = New-Object System.Xml.XmlWriterSettings
+    $xmlWriterSettings.Indent = $false
+    $xmlWriterSettings.OmitXmlDeclaration = $true
+
+    $xmlWriter = [System.Xml.XmlWriter]::Create($stringWriter, $xmlWriterSettings)
+    $XDoc.WriteTo($xmlWriter)
+    $xmlWriter.Close()
+
+    $output = $stringWriter.ToString()
+
+    If ($output.EndsWith("`r`n"))
+    {
+        $output = $output.Substring(0, $output.Length - 2)
+    }
+    ElseIf ($output.EndsWith("`n"))
+    {
+        $output = $output.Substring(0, $output.Length - 1)
+    }

```

Comment:
Maybe `[System.Xml.NewLineHandling]::None` will fix this

Reply:
Okay I just realized that it's VS Code that adds the newline whenever I open the file, not the xml API. So I don't need the NewLineHandling nor these if/else.

Reply:
Hmm they're still getting added even if I open the file in notepad, and even if I add that NewLineHandline.None setting. Oh well, I'll have to keep the if/else.

Reply:
This worked for me:
```
[Reflection.Assembly]::LoadWithPartialName("System.Xml.Linq")
$xdoc = [System.Xml.Linq.XDocument]::Load("in.csproj", [System.Xml.Linq.LoadOptions]::PreserveWhitespace)\

$xmlWriterSettings = New-Object System.Xml.XmlWriterSettings
$xmlWriterSettings.OmitXmlDeclaration = $true
$xmlWriterSettings.NewLineHandling = [System.Xml.NewLineHandling]::None

$xmlWriter = [System.Xml.XmlWriter]::Create("out.csproj", $xmlWriterSettings)
$xdoc.WriteTo($xmlWriter)
$xmlWriter.Close()
```

It was able to round trip the project file produced by .NET new without any changes.

I also found that the built-in XML support in PS worked well as shown in those other scripts.